### ISSUE TITLE
Processing Plugin for resolume using spoutcontrols crash if it the .txt file has more than 11 parameters.

### ISSUE THREAD

**Issue by jpupper** [leadedge/SpoutControls#2]
Grettings. 

Lately i´ve been updating an old processing-spout-plugin of mine called Spyrographics to work to the version of resolume  7.10 

You can check the proyect at : https://github.com/jpupper/Spyrographics

I found out that if you have more than 11 parameters define in the .txt file that has that kind of json for the values the resolume won´t open at all (my proyect has 20 parameters), this bug did not happen un resolume 5 with the old spoutControls version .

Here is an example : 

THIS MAKE THE PROGRAM CRASH : 
```

/*{
	"CREDIT": "SpoutControls - spout.zeal.co",
	"DESCRIPTION": "SpoutBox",
	"CATEGORIES": [
	"Parameter adjustment"
	],
	"INPUTS": [
		{
			"NAME": "Value1",
			"TYPE": "bool",
			"MIN": 0.00,
			"MAX": 1.00,
			"DEFAULT": 1.00
		},
		{
			"NAME": "Value2",
			"TYPE": "float",
			"MIN": 0.00,
			"MAX": 4.00,
			"DEFAULT": 1.00
		},
		{
			"NAME": "Value3",
			"TYPE": "float",
			"MIN": 0.00,
			"MAX": 4.00,
			"DEFAULT": 1.00
		},
		{
			"NAME": "Value4",
			"TYPE": "float",
			"MIN": 0.00,
			"MAX": 4.00,
			"DEFAULT": 1.00
		},
		{
			"NAME": "Value5",
			"TYPE": "float",
			"MIN": 0.00,
			"MAX": 4.00,
			"DEFAULT": 1.00
		},
		{
			"NAME": "Value6",
			"TYPE": "float",
			"MIN": 0.00,
			"MAX": 4.00,
			"DEFAULT": 1.00
		},
		{
			"NAME": "Value7",
			"TYPE": "float",
			"MIN": 0.00,
			"MAX": 4.00,
			"DEFAULT": 1.00
		},
		{
			"NAME": "Value8",
			"TYPE": "float",
			"MIN": 0.00,
			"MAX": 4.00,
			"DEFAULT": 1.00
		},
		{
			"NAME": "Value9",
			"TYPE": "float",
			"MIN": 0.00,
			"MAX": 4.00,
			"DEFAULT": 1.00
		},
		{
			"NAME": "Value10",
			"TYPE": "float",
			"MIN": 0.00,
			"MAX": 4.00,
			"DEFAULT": 1.00
		},
		{
			"NAME": "Value11",
			"TYPE": "float",
			"MIN": 0.00,
			"MAX": 4.00,
			"DEFAULT": 1.00
		},
		{
			"NAME": "Value12",
			"TYPE": "float",
			"MIN": 0.00,
			"MAX": 4.00,
			"DEFAULT": 1.00
		}
	],
}*/
```

THIS WORKS PERFECTLY JUST FINE : 

```
/*{
	"CREDIT": "SpoutControls - spout.zeal.co",
	"DESCRIPTION": "SpoutBox",
	"CATEGORIES": [
	"Parameter adjustment"
	],
	"INPUTS": [
		{
			"NAME": "Value1",
			"TYPE": "bool",
			"MIN": 0.00,
			"MAX": 1.00,
			"DEFAULT": 1.00
		},
		{
			"NAME": "Value2",
			"TYPE": "float",
			"MIN": 0.00,
			"MAX": 4.00,
			"DEFAULT": 1.00
		},
		{
			"NAME": "Value3",
			"TYPE": "float",
			"MIN": 0.00,
			"MAX": 4.00,
			"DEFAULT": 1.00
		},
		{
			"NAME": "Value4",
			"TYPE": "float",
			"MIN": 0.00,
			"MAX": 4.00,
			"DEFAULT": 1.00
		},
		{
			"NAME": "Value5",
			"TYPE": "float",
			"MIN": 0.00,
			"MAX": 4.00,
			"DEFAULT": 1.00
		},
		{
			"NAME": "Value6",
			"TYPE": "float",
			"MIN": 0.00,
			"MAX": 4.00,
			"DEFAULT": 1.00
		},
		{
			"NAME": "Value7",
			"TYPE": "float",
			"MIN": 0.00,
			"MAX": 4.00,
			"DEFAULT": 1.00
		},
		{
			"NAME": "Value8",
			"TYPE": "float",
			"MIN": 0.00,
			"MAX": 4.00,
			"DEFAULT": 1.00
		},
		{
			"NAME": "Value9",
			"TYPE": "float",
			"MIN": 0.00,
			"MAX": 4.00,
			"DEFAULT": 1.00
		},
		{
			"NAME": "Value10",
			"TYPE": "float",
			"MIN": 0.00,
			"MAX": 4.00,
			"DEFAULT": 1.00
		},
		{
			"NAME": "Value11",
			"TYPE": "float",
			"MIN": 0.00,
			"MAX": 4.00,
			"DEFAULT": 1.00
		}
	],
}*/
```

I´ve tryed copy and deleting several times the SpoutBox example and always crash. 

If you change the amount of parameters from 12 to 11 and open resolume it will work. If you close resolume after and change the parameters to 12 again the program will start BUT without the parameters on the plugin. 

It´s extremely wird

**Comment by leadedge** [comment]
I think this could be a problem with the plugin. 

The processing sketch, which is the sender, generates a control file in "C:\ProgramData\Spout" and the Freeframe plugin finds that file to set up the controls within Resolume.

Initially this file will not exist and the file in the plugin folder is used. That file should match the controls that the sketch creates. I have not had time to trace though the code to confirm it, but I think this could be the trouble. 

After running the plugin once, you should be able to find this file in "C:\ProgramData\Spout". If you re-start Arena, the controls will be re-set.

Does your original plugin work unchanged with the latest version of the Freeframe plugin?

**Comment by jpupper** [comment]
Is not a problem with my plugin . 

My plugin used to work perfect on resolume 5. 

Also Im testing it without my plugin, by just using the SpoutBox example by adding aditional values to the .txt file. 
The code of the SpoutBox remains unchange.

I´ve tryed checking the  file located at "C:\ProgramData\Spout" and i´ve put the parameters there without any luck.

I´ve tryed like 20,30 different combinations and its always the same. 

If you put more than 11 values on the .txt for some weird reason the resolume program won´t open. If you put 11 or less it will open everytime.

**Comment by leadedge** [comment]
I meant the Freeframe plugin, not your Processing sketch. 

I have created 12 controls in SpoutBox successfully using the stand-alone controller "SpoutController.exe" and I have reproduced the crash with Arena. This confirms that the problem is with the Freeframe plugin.

"SpoutController.exe" is part of the original SpoutControls installation. Copy this file to your sketch folder and run the sketch directly. You should see the multiple controls. You can use this to set up your sketch as required without loading Arena every time.

I have had a brief look at the plugin code but can't spot any possible cause. It will be difficult to debug because Arena closes immediately but I will try to test it somehow. It will take me a few days due to other commitments.

**Comment by leadedge** [comment]
Fortunately I had some time to look at this today. The culprit was - 

`Param::Ptr paramptr[10];`

Try [this version.](https://github.com/leadedge/leadedge.github.io/raw/downloads/SpoutControllerGL4_v2023.dll) Rename it to SpoutBox.dll or whatever is required.

It seems that Resolume now looks in subfolders of the Freeframe plugin directories like it used to, so you should be able to copy the dll into the SpoutBox folder. Please let me know if this works for you.

I had trouble with slow rendering if the sketch was moved off-screen. I have worked around the problem by moving it to the lower left corner and exposing a border-width of the window. Normally it will be hidden under the taskbar.

**Comment by jpupper** [comment]
Greetings. 

Thanks for the quick response. This new version of the spoutcontrols .dll is not crashing if you put more than 11 parameters. That seems to be fixed BUT is not opening the sketch (as the previous version of the .dll file does). 

Now I can open resolume, see the plugin with more than 11 eleven parameters BUT the sketch window won´t open. I´ve tried pressing the "show sender " button without any success. 

I´ve tryed putting 9 parameters and this new version that you send me and it won´t open either.

**Comment by jpupper** [comment]
Nevermind, i was putting the .dll outside the sketch folder in the "Extra Effects " root. But if you put the .dll inside the sketh folder works just alright with more than 10 parameters. Problem solve, thanks @leadedge ,i really love this developement you are so cool. thanks again, have a great day.

**Comment by leadedge** [comment]
Back after a short break. 
Good news that you have it working sucessfully. Apologies that folder detection had been removed in this plugin but I had not noticed the change. Thanks for the confirmation that Resolume detects Freeframe plugins in sub-folders. Now the dll can be in each sketch folder as originally intended.
I have updated everything and made a [new download](https://github.com/leadedge/leadedge.github.io/raw/downloads/SpoutControls_V2.023.zip).
